controller/TaskQueueWebApi.php

```diff
@@ -129,7 +131,49 @@ public function get()
     }
 
     /**
-     * @throws \common_exception_NotImplemented
+     * Gets the full url for a resource.
+     *
+     * @throws \Exception
+     */
+    public function getUrl()
+    {
+        if (!\tao_helpers_Request::isAjax()) {
+            throw new \Exception('Only ajax call allowed.');
+        }
+
+        try {
+            if (!$this->hasRequestParameter(self::PARAMETER_RESOURCE_URI)) {
+                throw new \common_exception_MissingParameter(self::PARAMETER_RESOURCE_URI, $this->getRequestURI());
+            }
+
+            $resource = new \core_kernel_classes_Resource($this->getRequestParameter(self::PARAMETER_RESOURCE_URI));
+
+            if ($resource->isClass()) {
+                $resource = new \core_kernel_classes_Class($resource);
+            }
+
+            if (!$resource->exists()) {
+                throw new \common_exception_NotFound('Resource "'. $this->getRequestParameter(self::PARAMETER_RESOURCE_URI). '" not found');
+            }
+
+            /** @var ResourceUrlBuilder $urlBuilder */
+            $urlBuilder = $this->getServiceLocator()->get(ResourceUrlBuilder::SERVICE_ID);
+
+            return $this->returnJson([
+                'success' => true,
+                'data' => $urlBuilder->buildUrl($resource)
+            ]);
+        } catch (\Exception $e) {
+            return $this->returnJson([
+                'success' => false,
+                'errorMsg' => $e instanceof \common_exception_UserReadableException ? $e->getUserMessage() : $e->getMessage(),
```

Comment:
For me it looks a little weird.
In my opinion, it is better to split to separate catch blocks. It would be more readable

Reply:
I do not really find it weird. Furthermore, it is used the same way in the other methods of this controller, so I would stay consequent.

Reply:
`instanceof ` is a codesmell
Sometimes it could be the only solution, but it is not that case.

Reply:
common_exception_UserReadableException is the interface to check to see if the exception has a description that is safe to send to the user, thought it might be more elegant to catch it in a seperate catch block
Sending the system exception message to an enduser however is unsafe and might contain sensitive informations